### PR TITLE
[SPARK-44432][SS][CONNECT] Terminate streaming queries when a session times out in Spark Connect

### PR DESCRIPTION
**Pull request by bogao007**
### What changes were proposed in this pull request?

Spark Connect server keeps a session id at connect server alive as long as a streaming query is alive. This may not be desirable. If the client disconnects long enough and times out in server, we should let the session mapping to be removed and terminate the streaming queries.

### Why are the changes needed?

As a Spark Connect semantic, we want to cleanup running queries when client is disconnected/timed out for a long time that the related session mapping is removed from the server.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Updated existing unit test to include new test case.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/artifact/SparkConnectArtifactManager.scala [hunk 1]
```diff
@@ -183,6 +183,9 @@ class SparkConnectArtifactManager(sessionHolder: SessionHolder) extends Logging
 
     // Clean up artifacts folder
     FileUtils.deleteDirectory(artifactPath.toFile)
+
+    // Clean up running queries
+    SparkConnectService.streamingSessionManager.cleanupRunningQueries(sessionHolder)
```
Comment (HyukjinKwon) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/artifact/SparkConnectArtifactManager.scala:188]: I actually think it has to be placed under `expireSession`. This class is really all about the artifacts you add, and strning query cleanup is irrelevant.

Comment (bogao007) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/artifact/SparkConnectArtifactManager.scala]: Thanks @HyukjinKwon, updated.

**Review by WweiL — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

**Comment by rangadi** [comment]
How is this tested? I think it will not timeout the session on the server at all because we have an explicit feature to keep the session mapping alive (implemented in #40937). 
 Even if the  client disappear, session would be kept alive and thus the queries will  not be cleaned up. 
 I think we need to remove most of  #40937.

File: connector/connect/server/src/test/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCacheSuite.scala [hunk 2]
```diff
@@ -95,7 +95,9 @@ class SparkConnectStreamingQueryCacheSuite extends SparkFunSuite with MockitoSug
     // Query is returned when correct session is used
     assert(sessionMgr.getCachedQuery(queryId, runId, mockSession).contains(mockQuery))
 
-    // Stop the query.
+    // Cleanup the query and verify if stop() method has been called.
+    sessionMgr.cleanupRunningQueries(sessionHolder)
```
Comment (rangadi) [inline connector/connect/server/src/test/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCacheSuite.scala:99]: This is not testing the desired functionality. It is just testing this cleanup method, not how it is used. we want the session to timeout normally and associated queries should be closed. 

Can we add a test for that? Also, we should manually verify this.

**Pull request closed without merging**